packages/marshal/src/types.js

```diff
@@ -2,70 +2,317 @@
 /// <reference path="extra-types.d.ts" />
 
 /**
- * @typedef { "bigint" | "boolean" | "null" | "number" | "string" | "symbol" | "undefined" | "copyArray" | "copyRecord" | "copyError" | "promise" | "presence" } PassStyle
- * TODO "presence" above should indirect through REMOTE_STYLE to prepare
- * for changing it to "remotable"
+ * @typedef { "undefined" | "null" |
+ *   "boolean" | "number" | "bigint" | "string" | "symbol" |
+ *   "copyArray" | "copyRecord" | "copyError" | "remotable" | "promise"
+ * } PassStyle
+ *
+ * TODO: We need to add a binary blob type to the PassStyles above and the
+ * LeafData below, consisting of an uninterpreted sequence of octets (8 bit
+ * bytes). The ideal JS representation would be a hardened `Uint8Array` except
+ * that Uint8Arrays cannot be frozen. Instead the likely JS representation
+ * will be a hardened ArrayBuffer.
  */
 
-// TODO declare more precise types throughout this file, so the type system
-// and IDE can be more helpful.
+/**
+ * @typedef { void | null | boolean | number | bigint | string | symbol
```

Comment:
`void` is only to indicate that a return value must not be depended on.  The type you're reaching for is `undefined`:

```suggestion
 * @typedef { undefined | null | boolean | number | bigint | string | symbol
```

Reply:
It used to say `undefined` but that was producing weird errors in code that was using the variable name `undefined`. I never did understand what it was complaining about. So I'll try it. If I still get such errors I'll post them here.